oshi-core/src/main/java/oshi/util/ConfigUtil.java

```diff
@@ -0,0 +1,164 @@
+     * 
+     * @param key The property key
+     * @return The property value or {@code null} if not found
+     */
+    public static String get(String key) {
+        return configuration.getProperty(key);
+    }
+
+    /**
+     * Get the {@code String} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the default if not found
+     */
+    public static String get(String key, String def) {
+        return configuration.getProperty(key, def);
+    }
+
+    /**
+     * Get the {@code Integer} property associated with the given key.
+     * 
+     * @param key The property key
+     * @return The property value or {@code 0} if not found
+     */
+    public static int getInteger(String key) {
+        return getInteger(key, 0);
+    }
+
+    /**
+     * Get the {@code Integer} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the default if not found
+     */
+    public static int getInteger(String key, int def) {
+        String value = configuration.getProperty(key);
+        return value == null ? def : Integer.parseInt(value);
+    }
+
+    /**
+     * Get the {@code Double} property associated with the given key.
+     * 
+     * @param key The property key
+     * @return The property value or {@code 0.0} if not found
+     */
+    public static Double getDouble(String key) {
+        return getDouble(key, 0.0);
+    }
+
+    /**
+     * Get the {@code Double} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the default if not found
+     */
+    public static Double getDouble(String key, double def) {
+        String value = configuration.getProperty(key);
+        return value == null ? def : Double.parseDouble(value);
+    }
+
+    /**
+     * Get the {@code Boolean} property associated with the given key.
+     * 
+     * @param key The property key
+     * @return The property value or {@code false} if not found
+     */
+    public static boolean getBoolean(String key) {
+        return getBoolean(key, false);
+    }
+
+    /**
+     * Get the {@code Boolean} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the default if not found
+     */
+    public static boolean getBoolean(String key, boolean def) {
+        String value = configuration.getProperty(key);
+        return value == null ? def : Boolean.parseBoolean(value);
+    }
+
+    /**
+     * Set the given property, overwriting any existing value.
+     * 
+     * @param key The property key
+     * @param val The new value
+     */
+    public static void set(String key, Object val) {
```

Comment:
I like the `toString()` approach here.  Will it throw an exception on `val = null`?   Do we want to allow null and invoke `remove()` in that case?

Reply:
I can't think of a reason to undefine a property, but calling `remove()` is probably more useful than throwing `NullPointerException`.

Reply:
My use case was that by clearing custom user behavior, we would restore default behavior.  Like if I set a custom WMI timeout and then wanted to remove my custom application, it would automatically revert to the infinite, without me explicitly having to *set* infinite.

I think a common use case for property files is for users to comment-out properties that aren't used/overridden, implying those would be the defaults... so in the "normal" situation there actually wouldn't be any properties on the object, and allowing removal helps restore it to its original state.

Reply:
Calling `remove()` won't restore the global default though.

Reply:
Remove will remove the property; so the behavior when `get()` is called will go back to "returning `null`" and restoring whatever the original "default" behavior was.

In the WMI case, with the "factory" implementation, the new global will be set on every new WMI initialization, and actually change behavior.  In the proc case, since that value is never read again, it won't change anything.